eodhp_utils/aws/egress_classifier.py

```
@@ -0,0 +1,174 @@
+
+        self.aws_ip_data_url = url
+        self.current_region = current_region
+        self.fallback_file = cache_file
+
+        self._build_tree()
+
+    def _build_tree(self):
+        ip_data = (
+            self._get_ip_data_from_aws()
+            or self._get_cached_ip_data()
+            or self._get_bundled_ip_data()
+        )
+
+        self.current_tree, self.aws_tree = self.build_trees(ip_data, self.current_region)
+        self.load_time = time.time()
+
+    def _get_ip_data_from_aws(self) -> Optional[dict]:
+        logger.debug("Fetching AWS IP address file from %s", self.aws_ip_data_url)
+
+        try:
+            response = requests.get(self.aws_ip_data_url)
+            response.raise_for_status()
+
+            logger.info("Fetched AWS IP address file from %s", self.aws_ip_data_url)
+            result_data = response.json()
+            assert isinstance(result_data, dict)
+
+            self._write_cache_file(response.content)
+
+            return result_data
+        except requests.RequestException:
+            logger.exception("Failed to fetch AWS IP address file at %s", self.aws_ip_data_url)
+            return None
+
+    def _get_cached_ip_data(self) -> Optional[dict]:
+        try:
+            with open(self.fallback_file, "r") as f:
+                ip_data = json.load(f)
+                return ip_data if isinstance(ip_data, dict) else None
+        except (json.JSONDecodeError, IOError):
+            logger.warning(
+                "Invalid or inaccessible AWS IP ranges cache file, %s. Ignoring.",
+                self.fallback_file,
+            )
+            return None
+
+    def _get_bundled_ip_data(self) -> dict:
+        bundled_file = dirname(__file__) + os.sep + "ip-ranges.json"
+        with open(bundled_file, "r") as f:
+            raw = f.read()
+            return json.loads(raw)
+
+    def _write_cache_file(self, content: bytes):
+        if not os.access(dirname(self.fallback_file), 0):
```

Should we also consider checking if that is a directory or not and whether it also has the write permission?

---

If this directory doesn't exist then it will usually mean that the /var/cache/eodhp/eodhp-utils/ directory hasn't been mounted into the container or that we're not running in a container. If it exists but it's a file or is not writeable then it has been configured incorrectly. That's something that should be fixed so I think it should let the exception from `open()` be raised.

Arguably it should raise an exception if it's running in a container and this directory doesn't exist but that would break development environments without something more complex being added.